runners/core-java/src/main/java/org/apache/beam/runners/core/SplittableParDo.java

```diff
@@ -351,7 +366,9 @@ public void processElement(ProcessContext context) {
     private StateTag<Object, ValueState<RestrictionT>> restrictionTag;
 
     private final DoFn<InputT, OutputT> fn;
-    private final Coder<? extends BoundedWindow> windowCoder;
+    private final Coder<InputT> elementCoder;
+    private final Coder<RestrictionT> restrictionCoder;
+    private final WindowingStrategy<InputT, ?> inputWindowingStrategy;
```

Comment:
What are you using these for? You need them for creating your `StateTags`, but everything else is unused.

Additionally, can runners not obtain these (if necessary) from the actual graph/application?